src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -239,6 +241,32 @@ public void addExternalTrigger(ExternalTrigger external_trigger) {
 		ext_triggers.put(external_trigger.getID(), external_trigger.getName());
 	}
 
+	@SuppressWarnings("unused") public boolean hasDependency(String procedureName, String dependencyName) {
+		ModElement me = mcreator.getWorkspace().getModElementByName(procedureName);
+		return me != null && me.getGeneratableElement() instanceof Procedure procedure && procedure.getDependencies()
+				.stream().anyMatch(e -> e.getName().equals(dependencyName));
+	}
+
+	@SuppressWarnings("unused") public String getDependencyType(String procedureName, String dependencyName) {
+		ModElement me = mcreator.getWorkspace().getModElementByName(procedureName);
+		if (me != null && me.getGeneratableElement() instanceof Procedure procedure) {
+			Optional<Dependency> dependency = procedure.getDependencies().stream()
+					.filter(e -> e.getName().equals(dependencyName)).findFirst();
+			if (dependency.isEmpty())

```

Comment:
I'd replace this check with `if (dependency.isPresent())` (similar to what it was before)

Reply:
:heavy_check_mark: Fixed